kolibri/core/assets/src/vue/textbox.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+
+  <ui-textbox
+    @focus="$emit('focus')"
+    @input="updateText()"
+    v-model = "currentText"
+    :disabled="disabled"
+    :placeholder = "placeholder"
+    :label = "label"
+    :aria-label = "ariaLabel"
+    :autocomplete = "autocomplete"
+    :autofocus = "autofocus"
+    :required = "required"
+    :type = "type"
+    :invalid="invalid"
+    class="koli-textbox">
+  </ui-textbox>
+
+</template>
+
+
+<script>
+
+  module.exports = {
+    name: 'KolibriTextBox',
+    props: {
+      disabled: {
+        type: Boolean,
+      },
+      autofocus: {
+        type: Boolean,
+      },
+      required: {
+        type: Boolean,
+      },
+      invalid: {
+        type: Boolean,
+      },
+      value: {
+        type: String,
+      },
+      placeholder: {
+        type: String,
+      },
+      label: {
+        type: String,
+      },
+      ariaLabel: {
+        type: String,
+        // enforcing accessibility
+        required: !!this.label,
```

Comment:
Interesting idea, but I'm a little confused by the logic/intent. Could you elaborate?